### PR TITLE
Rename macros and build options NAMEDTENSOR_ENABLED to BUILD_NAMEDTENSOR

### PR DESCRIPTION
**Pull request by xuhdev**
Currently the build system accepts USE_NAMEDTENSOR from the environment
variable and turns it into NAMEDTENSOR_ENABLED when passing to CMake.
This discrepancy does not seem necessary and complicates the build
system. The naming of this build option is also semantically incorrect
("BUILD_" vis-a-vis "USE_").  This commit eradicate this issue before it
is made into a stable release.

The support of NO_NAMEDTENSOR is also removed, since PyTorch has been
quite inconsistent about "NO_*" build options.

---

Note: All environment variables with their names starting with `BUILD_` are currently automatically passed to CMake with no need of an additional wrapper.

### CODE REVIEW

**Comment by xuhdev** [comment]
Besides all the renaming changes, three lines in `tools/setup_helpers/cmake.py` and `tools/setup_helpers/env.py` that do build option conversion are deleted.

**Comment by ezyang** [comment]
@zou3519 can you check on this and approve if you think it is good?

**Comment by zou3519** [comment]
@xuhdev could you please add `[namedtensor ci]` to your commit message so that the named tensor CI job runs on this pull request?

**Comment by zou3519** [comment]
@xuhdev I landed a few things this morning that added some more uses of the macro, so you'll need to rebase. I'll hold off on landing more changes for a day or two so that we can get this PR in.

**Comment by ezyang** [comment]
@zou3519 feel free to commandeer this patchset if you think that would be easier

**Comment by xuhdev** [comment]
@zou3519 rebased and added the ci command

**Review by zou3519 — APPROVED** [review]
Looks fine, just waiting for tests to pass

File: aten/src/THC/generic/THCTensorMathPointwise.cu
Comment (zou3519) [inline aten/src/THC/generic/THCTensorMathPointwise.cu]: This is wrong; I think you intended to change NAMEDTENSOR_ENABLED to BUILD_NAMEDTENSOR.

File: aten/src/THC/generic/THCTensorMathPointwise.cu
Comment (zou3519) [inline aten/src/THC/generic/THCTensorMathPointwise.cu]: same here

File: aten/src/THC/generic/THCTensorMathPointwise.cu
Comment (zou3519) [inline aten/src/THC/generic/THCTensorMathPointwise.cu]: same here

**Review by zou3519 — CHANGES_REQUESTED** [review]

File: aten/src/THC/generic/THCTensorMathPointwise.cu
Comment (xuhdev) [reply aten/src/THC/generic/THCTensorMathPointwise.cu]: Right. I probably messed it up during conflict resolving.

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**
